src/fraction/unicode_fromto_str.rs

```diff
@@ -0,0 +1,589 @@
+        struct UnicodeDisplay<'a, T: Clone + Integer>(&'a GenericFraction<T>);
+        impl<'a, T> fmt::Display for UnicodeDisplay<'a, T>
+        where
+            T: Clone + Integer + fmt::Display,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                match &self.0 {
+                    GenericFraction::NaN => write!(f, "NaN"),
+                    GenericFraction::Infinity(s) => {
+                        write!(f, "{}∞", s)
+                    }
+                    GenericFraction::Rational(s, r) => {
+                        write!(f, "{}{}", s, r.numer())?;
+                        if !r.denom().is_one() {
+                            write!(f, "⁄{}", r.denom())?;
+                        }
+                        Ok(())
+                    }
+                }
+            }
+        }
+        UnicodeDisplay(self)
+    }
+
+    /// Display the fraction using FRACTION SLASH '\u{2044}' '⁄' as a mixed fraction e.g. "1⁤1⁄2"
+    /// Will put INVISIBLE PLUS '\u{2064}' as a separator '⁤'
+    /// If you have font support, this is the way Unicode wants to display fractions
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).display_unicode_mixed()),
+    ///   "1⁄2"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).display_unicode_mixed()),
+    ///   "1⁤1⁄2"
+    /// );
+    /// ```
+    pub fn display_unicode_mixed(&self) -> impl Display + '_ {
+        struct S<'a, T: Clone + Integer + fmt::Display>(&'a GenericFraction<T>);
+        impl<'a, T> fmt::Display for S<'a, T>
+        where
+            T: Clone + Integer + fmt::Display + From<u8>,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                match &self.0 {
+                    GenericFraction::Rational(s, r) if r.fract() != *r && !r.denom().is_one() => {
+                        write!(
+                            f,
+                            "{}{}\u{2064}{}⁄{}",
+                            s,
+                            r.trunc().numer(),
+                            r.fract().numer(),
+                            r.fract().denom()
+                        )
+                    }
+                    _ => write!(f, "{}", self.0.display_unicode()),
+                }
+            }
+        }
+        S(self)
+    }
+
+    /// Display the fraction using super/subscript
+    /// This will look OK without font support
+    /// ```
+    /// use fraction::Fraction;
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(1u8,2u8).display_supsub()),
+    ///   "¹/₂"
+    /// );
+    /// assert_eq!(
+    ///   format!("{}", Fraction::new(3u8,2u8).display_supsub()),
+    ///   "³/₂"
+    /// );
+    /// ```
+    pub fn display_supsub(&self) -> impl Display + '_ {
+        struct S<'a, T: Clone + Integer + fmt::Display>(&'a GenericFraction<T>);
```

Comment:
Similar here, would be neat to expose a `UnicodeDisplay` struct with a method `supsub()`? so that would become `fraction.get_unicode_display().supsub()`.